sycl/include/CL/sycl/detail/program_impl.hpp

```diff
@@ -234,7 +238,11 @@ class program_impl {
           detail::getSyclObjImpl(Context)->getHandleRef(),
           Devices.size(), Devices.data(), LinkOptions.c_str(),
           1, &Program, nullptr, nullptr, &Err), Err));
-      PI_CHECK_THROW(Err, compile_program_error);
+
+      if (Err != PI_SUCCESS) {
+        throw compile_program_error("Program link error:\n" +
```

Comment:
Very interesting, the SYCL specification defines sycl::link_program_error type of exception with the description: "Error while linking the SYCL kernel to aSYCL device.", but in the descriptions of sycl::program methods it always says sycl::compile_program_error should be returned even in case of link operation.
@mkinsner FYI